node/core/pvf/common/src/worker/security.rs

```diff
@@ -186,3 +186,98 @@ pub mod landlock {
 		}
 	}
 }
+
+// TODO: Add a check for whether seccomp is supported and warn if not, like we do for landlock.
+//       Ideally we would have <https://github.com/rust-vmm/seccompiler/issues/51>.
+//       In lieu of that we can try restricting a dummy thread like we do for landlock.
+/// Process sandboxing using seccomp, a Linux facility for blocking syscall execution.
+///
+/// Currently we only block two syscalls, `socket` and `socketpair`, to effectively disable
+/// networking.
+///
+/// # Action on Caught Syscall
+///
+/// TODO: Update with the action and explain why given action was chosen.
+#[cfg(target_os = "linux")]
```

Comment:
Hmm.... we probably also want to gate this on architecture (`target_arch = "x86_64"`).

Technically seccompiler should also support aarch64, but without us auditing aarch64's list of syscalls (which do differ between architectures) and testing that it actually works there's a chance that it'd be broken/incomplete/insecure.

Reply:
seccompiler supports aarch64 and I didn't think there'd be any complication with the two socket creation syscalls. I considered gating on these two architectures, but seccompiler already does it so it would already be a compiler error.

What do you think about my idea of having a job that downloads the syscall table for these two architectures, from the latest kernel version, and checks for unexpected changes? I think we would need eventually it but it should not be a blocker for this PR as the scope is so small.

Reply:
> seccompiler supports aarch64 and I didn't think there'd be any complication with the two socket creation syscalls.

Yeah, probably, but we're going to extend this, right? I'd rather we either go all the way and properly support it, or not support it at all.

> but seccompiler already does it so it would already be a compiler error on an unsupported arch.

It'd be nice to handle this gracefully on unsupported architectures (essentially treat it as being compiled on an unsupported OS) instead of failing to compile at all.

> What do you think about my idea of having a job that downloads the syscall table for these two architectures, from the latest kernel version, and checks for unexpected changes?

You mean extract it from the Linux sources? I'm not entirely sure that it can be done with 100% reliability, but the scripts that are floating around which do that do seem like they can grab all of them from what I can see (at least for amd64). I don't think it's worth having a job like this to run for every PR, but a nightly job which runs once each day and pings us if there are any changes - sure, why not.

Reply:
> Would this be used to manually check the newly added syscalls, to see if there's a newly introduced network syscall?

Yeah, and my idea was to review new syscalls and determine if they should be blacklisted (I figure 99% of the time no action would be required)